sqlx-core/src/connection.rs

```diff
@@ -27,6 +27,19 @@ pub trait Connection: Send {
     ///
     /// Returns a [`Transaction`] for controlling and tracking the new transaction.
     fn begin(&mut self) -> BoxFuture<'_, Result<Transaction<'_, Self::Database>, Error>>
+    where
+        Self: Sized,
+    {
+        Self::begin_with(self, Default::default())
+    }
+
+    /// Begin a new transaction or establish a savepoint within the active transaction.
+    ///
+    /// Returns a [`Transaction`] for controlling and tracking the new transaction.
+    fn begin_with(
+        &mut self,
+        options: <<Self::Database as Database>::TransactionManager as TransactionManager>::Options,
+    ) -> BoxFuture<'_, Result<Transaction<'_, Self::Database>, Error>>
```

Comment:
I'm sorry, I should have looked at this PR before releasing 0.6.0 because this is a breaking change.

While it's _unlikely_ that any implementation of this trait exists outside SQLx, since we haven't sealed any of them it's not impossible.